Kickstarter-iOS/Library/SharedFunctionsTests.swift

```diff
@@ -115,7 +115,24 @@ internal final class SharedFunctionsTests: XCTestCase {
     XCTAssertEqual(backedReward, reward(from: backing, inProject: project))
   }
 
-  func testRewardFromBackingWithProject_WhenRewardNotPresent() {
+  func testRewardFromBackingWithProject_WhenRewardNotPresent_NoRewardPresent() {
+    let backing = Backing.template
+      |> Backing.lens.reward .~ nil
+      |> Backing.lens.rewardId .~ 123
+    let noReward = Reward.noReward
+      |> Reward.lens.minimum .~ 10
+    let project = Project.template
+      |> Project.lens.personalization.backing .~ backing
+      |> Project.lens.rewards .~ [
+        noReward, // No Reward "reward" is available
+        Reward.template
+          |> Reward.lens.id .~ 456
+      ]
+
+    XCTAssertEqual(noReward, reward(from: backing, inProject: project))
+  }
+
+  func testRewardFromBackingWithProject_WhenRewardNotPresent_NoRewardNotPresent() {
```

Comment:
Is it possible to have a project with no `NoReward`? 🤔

Reply:
According to @ifbarrera , it's probable that there's no project with no `NoReward`, but this test would cover this weird case if it happens.